### DISCUSSION TITLE
Clap for Delphi

### DISCUSSION THREAD

**Discussion by RuudErmers** [free-audio/clap#208] [Q&A] [answered]
I do not know if anyone was able to create a clap plugin with Delphi. I am trying to and I am almost there. 
(I created VST2, 3 and Fruity Plugs in the past, so I'd like to extend it to Clap). However, I have a problem with exporting clap_entry. This should be a const record, but Delphi can't export that!
E.g:

VAR n:integer;
exports n;   // no problem

const n:integer=3;
exports n; // not posible

VAR n:integer=3;
exports n; // not possible.

VAR clap_entry: Tclap_plugin_entry =
(
   clap_version :(major:1; minor:0; revision:17);
   init : entry_init;
   deinit : entry_deinit;
   get_factory : entry_get_factory;
);

exports clap_entry; // not possible....

What now ?

**Comment by Bremmers** [comment, marked as answer]
You can rename your clap_entry variable to plugin_entry, and then do this:

var
  plugin_entry: Tclap_plugin_entry;

exports plugin_entry name clap_entry;

begin
  plugin_entry.clap_version := CLAP_VERSION;
  plugin_entry.init := entry_init;
  plugin_entry.deinit := entry_deinit;
  plugin_entry.get_factory := entry_get_factory;
end.

**Reply by RuudErmers** [reply]
Thanks, that at least compiles. Tomorrow is the big day, and I'll check it out.. Thanks for the headers, BTW!.
(also I did not do what you suggested, but rather only wrote the code between begin...end. I was under the impression that no code was executed when loading that clap_entry, but as I wrote , hope to test tomorrow! 
(O btw btw: clap is a very nice design)

**Reply by RuudErmers** [reply]
It works. I thought you had to define a constant, so that Clap could read it without loading the plugin. But it does and you can e.g. call some functions to fill the returned clap.entry.
